jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/Content.java

```diff
@@ -561,7 +624,7 @@ static void write(Sink sink, boolean last, String utf8Content, Callback callback
      * to release the {@code ByteBuffer} back into a pool), or the
      * {@link #release()} method overridden.</p>
      */
-    public interface Chunk extends Retainable
+    public interface Chunk extends RetainableByteBuffer
```

Comment:
Chunk already has a read/write API as everything is currently done via `getByteBuffer` so we have the possibility of  `chunk.getByteBuffer().put(...)`.    If fact I can't see that we are very rigorous with ensuring the buffer within a chunk is actually read-only, so we should definitely check that regardless of this PR.

The current `ContentSourceByteBuffer` does `accumulator.copyBuffer(chunk.getByteBuffer())`, forcing a data copy when we could just retain.   This PR allows this code to become just: `accumulator.append(chunk)`, with zero copy.       The other alternative is to build a RBB of the chunk passing the `ByteBuffer` and the chunk as `Retainable`, but that is just creating garbage to try to make chunk look read-only, when it will never have read-only API because it exposes the `ByteBuffer` API.

Perhaps we should go the extra step and create a  `RBB.Mutable`, but this is kind of meaningless unless we can completely hide `ByteBuffer` 

See more discussion in the general comments below...